components/org.wso2.carbon.identity.data.publisher.audit.user.operation/src/main/java/org/wso2/carbon/identity/data/publisher/audit/user/operation/model/UserData.java

```diff
@@ -80,60 +75,52 @@ public void setTenantDomain(String tenantDomain) {
         this.tenantDomain = tenantDomain;
     }
 
-    public Object getCredentials() {
-        return credentials;
+    public String getNewRoleList() {
+        return newRoleList;
     }
 
-    public void setCredentials(Object credentials) {
-        this.credentials = credentials;
+    public void setNewRoleList(String newRoleList) {
+        this.newRoleList = newRoleList;
     }
 
-    public Object getOldCredentials() {
-        return oldCredentials;
+    public String getDeletedRoleList() {
+        return deletedRoleList;
     }
 
-    public void setOldCredentials(Object oldCredentials) {
-        this.oldCredentials = oldCredentials;
+    public void setDeletedRoleList(String deletedRoleList) {
+        this.deletedRoleList = deletedRoleList;
     }
 
-    public Object getNewCredentials() {
-        return newCredentials;
+    public String getClaims() {
+        return claims;
     }
 
-    public void setNewCredentials(String newCredentials) {
-        this.newCredentials = newCredentials;
+    public void setClaims(String claims) {
+        this.claims = claims;
     }
 
-    public String getActionHolder() {
-        return actionHolder;
-    }
-
-    public void setActionHolder(String actionHolder) {
-        this.actionHolder = actionHolder;
-    }
-
-    public String getNewRoles() {
-        return newRoles;
+    public String getProfile() {
+        return profile;
     }
 
-    public void setNewRoles(String newRoles) {
-        this.newRoles = newRoles;
+    public void setProfile(String profile) {
+        this.profile = profile;
     }
 
-    public String getDeletedRoles() {
-        return deletedRoles;
+    public String getActionHolder() {
+        return actionHolder;
     }
 
-    public void setDeletedRoles(String deletedRoles) {
-        this.deletedRoles = deletedRoles;
+    public void setActionHolder(String actionHolder) {
+        this.actionHolder = actionHolder;
     }
 
-    public String getProfile() {
-        return profile;
+    public long getActionTimestamp() {
+        return actionTimestamp;
     }
 
-    public void setProfile(String profile) {
-        this.profile = profile;
+    public void setActionTimestamp(long actionTimestamp) {
+        this.actionTimestamp = actionTimestamp;
     }
 
```

Comment:
The definition of this POJO has been changed. This is used to create events right ?
Will it not break any custom codes written extending publisher behavior.

Reply:
This is not currently used by IS analytics. I asked and confirmed that this can be changed.

All the removed POJOs are not used. (eg:- credentials will never be used since we will never publish credentials to analytics.)

Anyway changing this won't affect the events as long as the stream definitions are not changed.